### PR TITLE
SessionToken persistence implementation

### PR DESCRIPTION
**Pull request by tom-andersen**
Internal change to Firestore.

Prepare persistence of session token. This will be used in future work to support database name resuse.

### CODE REVIEW

File: Firestore/core/src/local/globals_cache.h [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+#ifndef FIRESTORE_CORE_SRC_LOCAL_GLOBALS_CACHE_H_
+#define FIRESTORE_CORE_SRC_LOCAL_GLOBALS_CACHE_H_
+
+#include "Firestore/core/src/nanopb/byte_string.h"
+
+namespace firebase {
+namespace firestore {
+namespace local {
+
+using nanopb::ByteString;
```
Comment (wu-hui) [inline Firestore/core/src/local/globals_cache.h:27]: Please use full path in header files. The reason is the files including the header will pick up these using statement as well, which might lead to surprises and conflicts down the road.

File: Firestore/core/src/local/leveldb_key.h [hunk 2]
```diff
@@ -768,6 +768,41 @@ class LevelDbNamedQueryKey {
   std::string name_;
 };
 
+/**
+ * A key in the globals table, storing the bundle Id for each entry.
```
Comment (wu-hui) [inline Firestore/core/src/local/leveldb_key.h:772]: "bundle Id" is copied from somewhere?

File: Firestore/core/src/local/leveldb_key.h [hunk 3]
```diff
@@ -768,6 +768,41 @@ class LevelDbNamedQueryKey {
...
+   * instance.
+   *
+   * @return true if the key successfully decoded, false otherwise. If false is
+   * returned, this instance is in an undefined state until the next call to
+   * `Decode()`.
+   */
+  ABSL_MUST_USE_RESULT
+  bool Decode(absl::string_view key);
+
+  /** The global name for this entry. */
```
Comment (wu-hui) [inline Firestore/core/src/local/leveldb_key.h:797]: "Global name" is vague here, can you please explain this better?

**Review by wu-hui — APPROVED** [review]
Some minor nits, please do fix them before merging.

LGTM in the meantime.

File: Firestore/CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,6 @@
 # Unreleased
 - [changed] Improve efficiency of memory persistence when processing a large number of writes. (#13572)
+- [changed] Prepare Firestore cache to support session token.
```
Comment (paulb777) [inline Firestore/CHANGELOG.md:3]: This change missed yesterday's code freeze. Please move to unreleased.

**Pull request merged**